xarray/backends/netCDF4_.py

```diff
@@ -527,6 +542,33 @@ def prepare_variable(
 
         return target, variable.data
 
+    def _build_and_get_enum(
+        self, var_name: str, dtype: np.dtype, enum_name: str, enum_dict: dict[str, int]
+    ) -> Any:
+        """
+        Add or get the netCDF4 Enum based on the dtype in encoding.
+        The return type should be ``netCDF4.EnumType``,
+        but we avoid importing netCDF4 globally for performances.
+        """
+        if enum_name not in self.ds.enumtypes:
```

Comment:
While working on https://github.com/h5netcdf/h5netcdf/pull/226 I've found that this test is not sufficient. netcdf4-python has different properties (`enumtypes`, cmptypes`, `vltypes`). If by chance a `vltype` has the same name as the new `enum_name` this will fail on type creation. We might need to check the other types too, or just wrap the type creation in a try/except.